proxy/proxy.go

```diff
@@ -315,134 +279,57 @@ func tryCatch(p func(), onErr func(err interface{})) {
-	return &http.Request{
-		Method:           r.Method,
-		URL:              cloneUrl(r.URL),
-		Proto:            r.Proto,
-		ProtoMajor:       r.ProtoMajor,
-		ProtoMinor:       r.ProtoMinor,
-		Header:           cloneHeader(r.Header),
-		Body:             &bodyBuffer{&bytes.Buffer{}},
-		ContentLength:    r.ContentLength,
-		TransferEncoding: r.TransferEncoding,
-		Close:            r.Close,
-		Host:             r.Host,
-		RemoteAddr:       r.RemoteAddr,
-		RequestURI:       r.RequestURI,
-		TLS:              r.TLS}
-}
-
-func cloneResponseMetadata(r *http.Response) *http.Response {
-	return &http.Response{
-		Status:           r.Status,
-		StatusCode:       r.StatusCode,
-		Proto:            r.Proto,
-		ProtoMajor:       r.ProtoMajor,
-		ProtoMinor:       r.ProtoMinor,
-		Header:           cloneHeader(r.Header),
-		Body:             &bodyBuffer{&bytes.Buffer{}},
-		ContentLength:    r.ContentLength,
-		TransferEncoding: r.TransferEncoding,
-		Close:            r.Close,
-		Request:          r.Request,
-		TLS:              r.TLS}
-}
-
-func (c *filterContext) ResponseWriter() http.ResponseWriter { return c.w }
-func (c *filterContext) Request() *http.Request              { return c.req }
-func (c *filterContext) Response() *http.Response            { return c.res }
-func (c *filterContext) MarkServed()                         { c.served = true }
-func (c *filterContext) Served() bool                        { return c.served || c.servedWithResponse }
-func (c *filterContext) PathParam(key string) string         { return c.pathParams[key] }
-func (c *filterContext) StateBag() map[string]interface{}    { return c.stateBag }
-func (c *filterContext) BackendUrl() string                  { return c.backendUrl }
-func (c *filterContext) OriginalRequest() *http.Request      { return c.originalRequest }
-func (c *filterContext) OriginalResponse() *http.Response    { return c.originalResponse }
-func (c *filterContext) OutgoingHost() string                { return c.outgoingHost }
-func (c *filterContext) SetOutgoingHost(h string)            { c.outgoingHost = h }
-
-func (c *filterContext) Serve(res *http.Response) {
-	res.Request = c.Request()
-
-	if res.Header == nil {
-		res.Header = make(http.Header)
-	}
-
-	if res.Body == nil {
-		res.Body = &bodyBuffer{&bytes.Buffer{}}
-	}
-
-	c.servedWithResponse = true
-	c.res = res
-}
-
-// creates an empty shunt response with the initial status code of 404
-func shunt(r *http.Request) *http.Response {
-	return &http.Response{
-		StatusCode: http.StatusNotFound,
-		Header:     make(http.Header),
-		Body:       &bodyBuffer{&bytes.Buffer{}},
-		Request:    r}
-}
-
 // applies all filters to a request
-func (p *Proxy) applyFiltersToRequest(f []*routing.RouteFilter, ctx *filterContext, onErr func(err interface{})) []*routing.RouteFilter {
-	var start time.Time
+func (p *Proxy) applyFiltersToRequest(f []*routing.RouteFilter, ctx *context) []*routing.RouteFilter {
```

Comment:
not sure actually. this one comes in pair with the applyFiltersToResponse, and both apply the filters that are passed in as args, but the response one doesn't necessarily receives all the filters, because of the shunting.